util/otbn_build.py

```diff
@@ -170,17 +170,49 @@ def main() -> int:
         out_elf = out_dir / (app_name + '.elf')
         call_otbn_ld(obj_files, out_elf, linker_script = args.linker_script)
 
+        # Use objcopy to create an ELF that can be linked into a RISC-V binary
+        # (to run on Ibex). This should set flags for all sections like rodata
+        # (since they're not executable on Ibex, nor does it make sense for
+        # Ibex code to manipulate OTBN data sections "in place"). We name them
+        # with a .otbn prefix, so end up with e.g. .rodata.otbn.text and
+        # .rodata.otbn.data.
+        #
+        # There are 4 symbols that we want to keep and link in, giving
+        # the start and end of imem and dmem. These will be relocated as part
+        # of the link, so they'll give addresses in the Ibex address space. So
+        # that the RISC-V binary can link multiple OTBN applications, we give
+        # them an application-specific prefix. (Note: This prefix is used in
+        # sw/device/lib/runtime/otbn.h: so needs to be kept in sync with that).
+        #
+        # The other symbols in the OTBN binary are not of interest to the
+        # RISC-V code, and they occupy a different namespace (referring to
+        # addresses in OTBN memories, rather than addresses in the rodata
+        # sections). Strip them out.
+        sym_pfx = '_otbn_app_{}_'.format(app_name)
+        kept_syms = ['_dmem_start', '_dmem_end',
+                     '_imem_start', '_imem_end']
+
         out_embedded_obj = out_dir / (app_name + '.rv32embed.o')
-        args = [
-            '-O',
-            'elf32-littleriscv',
-            '--prefix-symbols',
-            '_otbn_app_' + app_name + '_',
-            out_elf,
-            out_embedded_obj,
-        ]
+        args = (['-O', 'elf32-littleriscv',
+                 '--prefix-sections=.rodata.otbn',
+                 '--set-section-flags=*=alloc,load,readonly',
+                 '--prefix-symbols', sym_pfx] +
+                ['--keep-symbol={}{}'.format(sym_pfx, s) for s in kept_syms] +
```

Comment:
Is it possible that the option `--strip-all` is missing?

Reply:
Not just possible, but certain! :-D Thanks.